### PR TITLE
Optional protagonist

### PR DESCRIPTION
**Pull request by JackuB**
Follow up for https://github.com/apiaryio/drafter-npm/pull/15

Let's not install protagonist or show warning that peer dependency is missing. This will keep it really optional.

### CODE REVIEW

**Comment by JackuB** [comment]
Anyone knows how to resolve this?

```
make: Entering directory `/home/travis/build/apiaryio/drafter-npm/node_modules/protagonist/build'
  CXX(target) Release/obj.target/libdrafter/drafter/src/drafter.o
g++: error: unrecognized command line option ‘-std=c++14’
make: *** [Release/obj.target/libdrafter/drafter/src/drafter.o] Error 1
make: Leaving directory `/home/travis/build/apiaryio/drafter-npm/node_modules/protagonist/build'
gyp ERR! build error 
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,15 +1,13 @@
 # drafter-npm CHANGELOG
 
-## Master
+## 2.0.0 Breaking
 
-### Breaking
+- Protagonist is no longer required, but it's optional.
```
Comment (kylef) [inline CHANGELOG.md:5]: Seems you also dropped Node 4 support, this should be indicated in CHANGELOG too.

File: .travis.yml [hunk 2]
```diff
@@ -4,17 +4,13 @@ dist: trusty
...
-- if ! "$PROTAGONIST"; then sudo apt-get purge build-essential && sudo apt-get autoremove; fi
-
-before_script:
-- if ! "$PROTAGONIST" && [ -d "node_modules/protagonist" ]; then echo "Protagonist was successfully installed (should not have been)." && exit 1; fi
+script:
+- if [ -d "node_modules/protagonist" ]; then echo "Protagonist was installed (should not have been)." && exit 1; fi
+- npm test
+- npm i protagonist
+- if ! [ -d "node_modules/protagonist" ]; then echo "Protagonist was not installed." && exit 1; fi
+- npm test
```
Comment (kylef) [inline .travis.yml:13]: I think it would make sense to uninstall the drafter.js package before this test run. Otherwise if the code was changed / broken so that Protagonist was never used these tests would still pass due to drafter.js.

Or perhaps we can think about how we can assert that protagonist was actually used for parsing during the tests.

File: README.md [hunk 3]
```diff
@@ -73,6 +72,18 @@ Name                   | Description
 `requireBlueprintName` | Require parsed blueprints have a title (default: false)
 `generateSourceMap`    | Enable sourcemap generation (default: false)
 
+
+## Improving performance
+Optionally you can also install C++ bindings to [Drafter](https://github.com/apiaryio/drafter) - [Protagonist](https://github.com/apiaryio/protagonist).
+
+```bash
+npm i protagonist
```
Comment (kylef) [inline README.md:80]: I'd prefer if we used `install` instead of `i` especially in public facing documentation. It's clearer to users who are not too familiar with these commands.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,15 +1,13 @@
...
+
+## 1.2.0 Breaking
```
Comment (kylef) [inline CHANGELOG.md:7]: Seems like a mistake changing the 1.20 version.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,15 +1,13 @@
...
 
 - Drop support node < 4.0
 
 - Remove the option to select AST Type. The ouput will be only refract
 
-## 1.2.0
-
-### Enhancements
```
Comment (kylef) [inline CHANGELOG.md:13]: Why was this removed?

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,15 +1,13 @@
 # drafter-npm CHANGELOG
 
-## Master
+## 2.0.0 Breaking
```
Comment (kylef) [inline CHANGELOG.md:3]: The title should just be the version number here.

```
## 2.0.0
```

**Comment by kylef** [comment]
@JackuB I think you asked for help regarding the test failures but I forgot where so I'll reply here.

The problem is due to the `npm install protagonist` which is installing a newer version (pre release of next major). This is incompatible, and it requires a more modern compiler (that has to installed). To fix you will need to install the older version.

This brings up another question though, we can no longer lock the version of Protagonist that is used and users can come into problems where Protagonist and drafter.js API diverge (we currently control this based on version locking in package.json to compatible major versions).

**Comment by w-vi** [comment]
I think that it can be solved by checking the protagonist version you try to import. If it is different in major or maybe even minor issue warning and fallback on drafter.js.
